File: src/main/kotlin/LastStoneWeight.java
import java.util.PriorityQueue;

class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> heap = new PriorityQueue<>((o1, o2) -> {
            if (o1 > o2) return -1;
            else return 1;
        });
        for (int stone : stones) {
            heap.add(stone);
        }


        while (heap.size() > 1) {
            int a = heap.poll();
            int b = heap.poll();

            if (a != b) {
                heap.add(a - b);
            }
        }

        if (heap.isEmpty()) return 0;
        else return heap.peek();
    }
}

File: src/main/kotlin/LastStoneWeight.kt
import java.util.*

class Solution {
    fun lastStoneWeight(stones: IntArray): Int {
        val heap = PriorityQueue<Int>(stones.size) { a, b -> b - a }

//        stones.forEach { heap.add(it) }
        for (stone in stones) {
            heap.add(stone)
        }
        while (heap.size > 1) {
            val a = heap.poll()
            val b = heap.poll()
            if (a != b) heap.add(a - b)
        }

        return if (heap.isEmpty()) 0 else heap.peek()
    }
}